reposilite-plugins/webhook-plugin/src/main/kotlin/com/reposilite/plugin/webhook/listener/WebhookEvent.kt

```diff
@@ -0,0 +1,25 @@
+package com.reposilite.plugin.webhook.listener
+
+import com.reposilite.maven.Repository
+import com.reposilite.plugin.webhook.api.sendDiscordEmbed
+import com.reposilite.plugin.webhook.api.sendDiscordMessage
+import com.reposilite.storage.api.Location
+
+
+fun webhookEvent(by: String, gav: Location, repo: Repository) {
+    val webhookService = "discord" // Should be removed
+    val webhookType = "message" // Should be removed
+    if (webhookService === "discord") {
+        if (webhookType === "message") {
+            sendDiscordMessage(by, gav, repo)
+        } else if (webhookService === "embed") {
+            sendDiscordEmbed(by, gav, repo)
+        } else {
+            println("Reposilite Webhook: Invaild Webhook Type found, using embed.")
+            sendDiscordEmbed(by, gav, repo)
+        }
```

Comment:
You can use pattern matching with `when` statement in Kotlin

Reply:
Would a strategy be better?

Reply:
Probably, but it was a nice snippet to mention `when`, so you can now use it in the future for similar cases :)